mozci/console/commands/push.py

```diff
@@ -305,14 +314,160 @@ def handle(self):
+                fieldnames=[
+                    "branch",
+                    "push",
+                    "revision",
+                    "task_id",
+                    "created",
+                    "time_taken",
+                ],
+            )
+            writer.writeheader()
+            writer.writerows(stats)
+
+        self.line(f"<info>Written stats for {len(stats)} tasks in {output}</info>")
+
+    def parse_routes(self, routes):
+        """Find revision from task routes"""
+
+        def _match(route):
+            res = self.REGEX_ROUTE.search(route)
+            if res:
+                return res.groups()
+
+        # Extract branch+name+value from the routes
+        # and get 3 separated lists to check those values
+        branches, keys, values = zip(*filter(None, map(_match, routes)))
+
+        # We should only have one branch
+        branches = set(branches)
+        assert len(branches) == 1, f"Multiple branches detected: {branches}"
+
+        # Output single branch, revision and push id
+        data = dict(zip(keys, values))
+        assert "revision" in data, "Missing revision route"
+        assert "push" in data, "Missing push route"
+        return branches.pop(), data["revision"], int(data["push"])
+
+    def parse_task_status(self, task_status):
+        """Extract identification and time spent for each classification task"""
+
+        def date(x):
+            return datetime.datetime.strptime(x, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+        out = {
+            "task_id": task_status["status"]["taskId"],
+            "created": task_status["task"]["created"],
+            "time_taken": sum(
+                (date(run["resolved"]) - date(run["started"])).total_seconds()
+                for run in task_status["status"]["runs"]
+                if run["state"] == "completed"
+            ),
+        }
+        out["branch"], out["revision"], out["push"] = self.parse_routes(
+            task_status["task"]["routes"]
+        )
+        return out
+
+    def list_groups_from_hook(self, group_id, hook_id):
+        """List all decision tasks from the specified hook"""
+        hooks = taskcluster.Hooks(get_taskcluster_options())
+        fires = hooks.listLastFires(group_id, hook_id).get("lastFires", [])
+
+        # Setup CLI progress bar
+        progress = self.progress_bar(len(fires))
+        progress.set_format("verbose")
+
+        # Provide the decision task ID as it's the same value for group ID
+        for fire in fires:
+            yield fire["taskId"]
+
+            progress.advance()
+
+        # Cleanup progress bar
+        progress.finish()
+
+    def list_classification_tasks(self, group_id):
+        cache_key = f"perf/task_group/{group_id}"
+
+        # Check cache
+        tasks = config.cache.get(cache_key)
+        if tasks is None:
+            queue = taskcluster.Queue(get_taskcluster_options())
+            try:
+                tasks = queue.listTaskGroup(group_id).get("tasks", [])
```

Comment:
Do we need pagination or is it automatically handled by the taskcluster library?

Reply:
Argh pagination is not handled automatically

Reply:
Just added support for pagination